coalib/results/Diff.py

```diff
@@ -209,14 +211,18 @@ def __json__(self):
         """
         return self.unified_diff
 
-    def affected_code(self, filename):
+    def affected_code(self, filename=None):
         """
         Creates a list of SourceRange objects which point to the related code.
         Changes on continuous lines will be put into one SourceRange.
 
         :param filename: The filename to associate the SourceRange's to.
         :return:         A list of all related SourceRange objects.
         """
+        assert filename or self.filename
```

Comment:
A message like 
`assert filename or self.filename, "A filename is required create SourceRanges for"`
would go a long way to not confuse people later